### PR TITLE
Bubble Sort in TI-83 Basic

### PR DESCRIPTION
**Pull request by thecnoNSMB**
Hopefully this is all correct, I'm new at this. (Also, before you ask, TI-83 Basic doesn't allow for whitespace, so long as I know, and standard style is to leave everything at the same indentation level. Also, variable names can only be one character long.)

### CODE REVIEW

**Comment by julianschacher** [comment]
Hi, welcome to the AAA community and thanks for the contribution! That you implemented an algorithm in TI-83 is super cool.

There are a few things you need to include/watch out for:

### One
You need to add the following entry at the bottom of `book.json` file:
```
{
  "lang": "ti83b",
  "name": "TI-83 Basic"
}
```
(Don't forget to add a comma after the last curly bracket of the previous entry).


### Two
You committed your changes to your own `master` branch. That's not how you should do it, since it can easily cause problems.
For now: just don't update your master anymore (i.e. pulling anything from `upstream` (if you have that)).
For the future: Reset your master (or just download (clone) the algorithm-archive again) and create a feature branch, you can get a tutorial on how to do so (and many other cool things) in this playlist: https://www.youtube.com/playlist?list=PL5NSPcN6fRq2vwgdb9noJacF945CeBk8x
or in the How to Contribute guide, which will be added to the wiki in the following days.

If you have any questions or just want to engage with the community in general, feel free to join the discord: https://discord.gg/Pr2E9S6

**Comment by julianschacher** [comment]
Does it have a reason, the file name is `BUBLSORT.txt` and not `BUBBLESORT.txt`?
(I bet only 8 chars are allowed)

**Comment by leios** [comment]
Just want to pop in here and say this is really cool!

File: chapters/algorithms/bubble_sort/code/ti83basic/BUBLSORT.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+:"SUBMITTED BY GIBUS WEARING BRONY"
+:"LENGTH OF LIST"
+:dim(L1)→L
+:"IS SORTED?"
+:0→S
+:While not(S)
```
Comment (Butt4cak3) [inline chapters/algorithms/bubble_sort/code/ti83basic/BUBLSORT.txt]: I know this makes it faster and that some other language implementations in the book already use this sort of optimization, but I'd rather have you change this to 2 nested for loops.
```
For(A,1,L)
For(X,1,L-1)
```

The reason is that the chapter talks about the unoptimized version of BubbleSort and the Julia code, which is kind of the reference implementation, does it the same way.

**Comment by thecnoNSMB** [comment]
> You committed your changes to your own `master` branch. That's not how you should do it, since it can easily cause problems.
For now: just don't update your master anymore (i.e. pulling anything from `upstream` (if you have that)).

Thanks for letting me know. If I make additional changes on a branch, will those changes be reflected here? If not, how should I proceed? (I'll move forward with the requested changes once I know how.)

> Does it have a reason, the file name is `BUBLSORT.txt` and not `BUBBLESORT.txt`?
(I bet only 8 chars are allowed)

You are correct in that assumption.

**Comment by julianschacher** [comment]
Just make the changes like you did until now (And don't merge any other branch into your master for now). After this PR is done, we can do some cleanup.

**Comment by thecnoNSMB** [comment]
Okay, cool. Apologies for the inconvenience; I will make the requested changes.

**Comment by julianschacher** [comment]
No need to apology, committing to master branch is a super common mistake, and, if not messed up (which you didn't), not a critical one :)

**Comment by thecnoNSMB** [comment]
I've made all the requested changes. I have one question: in `bubble_sort.md`, it currently says `lang:"ti-83_basic"`. Should I change that to match the `"name": "TI-83 Basic"` in `book.json`?

**Comment by julianschacher** [comment]
This identifier you mentioned (`lang:"ti-83_basic"`) is needed for syntax highlighting. So it doesn't really matter anyway, because TI-83 Basic probably doesn't have syntax highlighting support. You can leave it as is.

What you added to the `book.json` is an entry for your language. This enables the language switching functionality you see at the top of each page. You then say with the following line, that all upcoming lines are about "TI-83 Basic":
```
{% sample lang="ti83b" %}
```
This `lang="ti83b"` is the same as what you added to the `book.json`. And in the end, the language switcher will show at the top, what you specified in the `book.json` (`"name": "TI-83 Basic"`) and will show the code (which you imported (`[import:2-13, lang:"ti-83_basic"](code/ti83basic/BUBLSORT.txt)`):
screen shot 2018-07-19 at 03 43 07_bubble_sort_ti-83_basic

**Comment by julianschacher** [comment]
I hope that answered a few questions :)

**Comment by thecnoNSMB** [comment]
Ah, neat! Thanks for the info. Let me know if any further changes need to be made.

**Comment by julianschacher** [comment]
There aren't further changes needed from my side - the book runs with the example included.
Maybe @Butt4cak3 has additional request tho, so we'll need to see, what he says (he's currently offline).

**Comment by Butt4cak3** [comment]
> Okay, cool. Apologies for the inconvenience
You absolutely don't have to apologize for anything. Change requests are completely normal. In fact, it was particularly pleasant to work on this PR because you did everything right. If you want to learn a bit about Git and GitHub, @julianschacher is working on Wiki pages about contribution and I started a [video series](https://www.youtube.com/playlist?list=PL5NSPcN6fRq2vwgdb9noJacF945CeBk8x) for new contributors.

I think we're already done here. The code is good and it renders properly in the book. Thanks for the contribution and welcome to the team!

**Comment by julianschacher** [comment]
Just wanted to leave a note in here, that the "How to Contribute" is finished: https://github.com/algorithm-archivists/algorithm-archive/wiki/How-to-Contribute

**Comment by jiegillet** [comment]
[Posting this for posterity. ](https://www.youtube.com/watch?v=iPpeWGzjPZY)

**Pull request merged**